alphabase/io/tempmmap.py

```diff
@@ -163,10 +163,16 @@ def array(shape: tuple, dtype: np.dtype, tmp_dir_abs_path: str = None) -> np.nda
         TEMP_DIR_NAME, f"temp_mmap_{np.random.randint(2**63, dtype=np.int64)}.hdf"
     )
 
-    with h5py.File(temp_file_name, "w") as hdf_file:
-        array = hdf_file.create_dataset("array", shape=shape, dtype=dtype)
-        array[0] = 0
-        offset = array.id.get_offset()
+    if isinstance(dtype, np.dtypes.ObjectDType):
```

Comment:
am I missing something or could this be refactored to

```python
with h5py.File(temp_file_name, "w") as hdf_file:
  array = hdf_file.create_dataset("array", shape=shape, dtype=dtype)
  array[0] = np.string_("") if isinstance(dtype, np.dtypes.ObjectDType) else 0
  offset = array.id.get_offset()
```
?

Reply:
no your not missing anything. Absolutely obvious seeing it like this xD

Reply:
done